backend/src/main/java/com/benine/backend/http/HttpController.java

```diff
@@ -26,27 +26,21 @@
    * Constructor, creates a new HttpController object.
    * @param address an internetsocet address indicating the ports for the server to listen to.
    * @param logger the logger to use to log to.
-   * @param serverController the server controller that contains
-   *                      the camera's which this server interacts with.
    * @param port number to connect to.
    */
-  public HttpController(String address, int port, Logger logger,
-                                                        ServerController serverController) {
-    this(createServer(address, port, logger), logger, serverController);
+  public HttpController(String address, int port, Logger logger) {
+    this(createServer(address, port, logger), logger);
   }
 
   /**
    * Constructor, creates a new HttpController object.
    * @param httpserver a server object.
    * @param logger the logger to use to log to.
-   * @param serverController the server controller that contains
-   *                      the camera's which this server interacts with.
    */
-  public HttpController(HttpServer httpserver, Logger logger, ServerController serverController) {
+  public HttpController(HttpServer httpserver, Logger logger) {
     this.logger = logger;
-    this.serverController = serverController;
     this.server = httpserver;
-
+    serverController = ServerController.getInstance();
```

Comment:
I think that if the serverController is a singleton we won't have to store it here as an attribute.